app/controllers/spree/admin/products_controller_decorator.rb

```diff
@@ -10,17 +11,17 @@ def get_suppliers
     @suppliers = Spree::Supplier.order(:name)
   end
 
-  def set_supplier
-    if try_spree_current_user.supplier?
-      @object.master.supplier = spree_current_user.supplier
-      @object.master.save
-    end
-  end
-
   # Scopes the collection to the Supplier.
   def supplier_collection
     if try_spree_current_user && !try_spree_current_user.admin? && try_spree_current_user.supplier?
-      @collection = @collection.joins(:suppliers).where('spree_suppliers.id = ?', spree_current_user.supplier_id)
+      @collection = @collection.joins(:suppliers).where('spree_suppliers.id = ?', try_spree_current_user.supplier_id)
```

Comment:
Line is too long. [117/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.